lxd/storage/drivers/driver_btrfs_volumes.go

```diff
@@ -1408,9 +1410,35 @@ func (d *btrfs) UnmountVolumeSnapshot(snapVol Volume, op *operations.Operation)
 	return forceUnmount(snapPath)
 }
 
-// VolumeSnapshots returns a list of snapshots for the volume (in no particular order).
+// VolumeSnapshots returns a list of snapshots for the volume (ordered by subvolume ID).
+// Since the subvolume ID is incremental, this also represents the order of creation.
 func (d *btrfs) VolumeSnapshots(vol Volume, op *operations.Operation) ([]string, error) {
-	return genericVFSVolumeSnapshots(d, vol, op)
+	args := []string{"subvolume", "list", GetPoolMountPath(vol.pool)}
```

Comment:
`btrfs subvolume list -s <path>` should only list snapshots according to the man page.

Reply:
In fact, it seems that `btrfs subvolume list` supports a bunch of sorting option, maybe one would do most of the work?

Reply:
> In fact, it seems that `btrfs subvolume list` supports a bunch of sorting option, maybe one would do most of the work?

Unfortunately, there's nothing more `btrfs subvolume list` could help us with here.